.buildkite/pipeline.yml

```diff
@@ -49,24 +49,48 @@ steps:
   #################
   - label: "🧪 UI Tests (iPhone)"
     command: |
+      echo "--- 📦 Downloading Build Artifacts"
       buildkite-agent artifact download build-products.tar .
       tar -xf build-products.tar
+
+      echo "--- :wrench: Fixing VM"
+      brew install openjdk@11
+      sudo ln -sfn /usr/local/opt/openjdk@11/libexec/openjdk.jdk /Library/Java/JavaVirtualMachines/openjdk-11.jdk
+
+      echo "--- :rubygems: Setting up Gems"
       bundle install
+
+      echo "--- :cocoapods: Setting up Pods"
       bundle exec pod install
+
+      echo "--- 🧪 Testing"
       rake mocks &
       bundle exec fastlane test_without_building name:WordPressUITests try_count:3 device:"iPhone 11" ios-version:"14.1"
     depends_on: "build"
-    agents:
-      os: "macOS"
+    env:
+      IMAGE_ID: xcode-12.5.1
+    <<: *use_plugins
 
   - label: "🧪 UI Tests (iPad)"
     command: |
+      echo "--- 📦 Downloading Build Artifacts"
       buildkite-agent artifact download build-products.tar .
       tar -xf build-products.tar
+
+      echo "--- :wrench: Fixing VM"
+      brew install openjdk@11
+      sudo ln -sfn /usr/local/opt/openjdk@11/libexec/openjdk.jdk /Library/Java/JavaVirtualMachines/openjdk-11.jdk
+
+      echo "--- :rubygems: Setting up Gems"
       bundle install
+
+      echo "--- :cocoapods: Setting up Pods"
       bundle exec pod install
+
+      echo "--- 🧪 Testing"
       rake mocks &
       bundle exec fastlane test_without_building name:WordPressUITests try_count:3 device:"iPad Air (4th generation)" ios-version:"14.1"
     depends_on: "build"
-    agents:
-      os: "macOS"
+    env:
+      IMAGE_ID: xcode-12.5.1
```

Comment:
Nitpick for future iterations: we should be able to extract either the whole `env` or the `IMAGE_ID` value to DRY, right?

Reply:
Absolutely – I just have no idea how to do that 😁 

Maybe @AliSoftware knows?

Reply:
https://github.com/wordpress-mobile/WordPress-iOS/pull/16929 😄